smarts/core/trap_manager.py

```diff
@@ -210,61 +245,52 @@ def largest_vehicle_plane_dimension(vehicle: Vehicle):
-            for v_id in sorted_vehicle_ids:
-                # Skip the capturing process if history traffic is used
-                if trap.mission.vehicle_spec is not None:
-                    break
-
-                if not trap.includes(v_id):
+            for vehicle_id in sorted_vehicle_ids:
+                if not trap.includes(vehicle_id):
                     continue
 
-                vehicle: Vehicle = vehicles[v_id]
+                vehicle: Vehicle = vehicles[vehicle_id]
                 point = vehicle.pose.point.as_shapely
 
                 if not point.within(trap.geometry):
                     continue
 
-                captures_by_agent_id[agent_id].append(
-                    (
-                        v_id,
-                        trap,
-                        replace(
-                            trap.mission,
-                            start=Start(vehicle.position[:2], vehicle.pose.heading),
-                        ),
-                    )
+                capture_by_agent_id[agent_id] = _CaptureState(
+                    ready_state=trap_condition,
+                    trap=trap,
+                    updated_mission=replace(
+                        trap.mission,
+                        start=Start(vehicle.position[:2], vehicle.pose.heading),
+                    ),
+                    vehicle_id=vehicle_id,
                 )
-                social_vehicle_ids.remove(v_id)
+                social_vehicle_ids.remove(vehicle_id)
                 break
+            else:
+                capture_by_agent_id[agent_id] = _CaptureState(
+                    ready_state=trap_condition,
+                    trap=trap,
+                )
 
         used_traps = []
-        for agent_id in (
-            sim.agent_manager.pending_agent_ids
-            | sim.agent_manager.pending_social_agent_ids
-        ):
-            trap = self._traps.get(agent_id)
+        for agent_id in pending_agent_ids:
+            capture = capture_by_agent_id[agent_id]
 
-            if trap is None:
+            if capture.default:
                 continue
 
-            if not trap.ready(sim.elapsed_sim_time):
+            if capture.trap is None:
                 continue
 
-            captures = captures_by_agent_id[agent_id]
+            if not capture.trap.ready(sim.elapsed_sim_time):
+                continue
 
             vehicle: Optional[Vehicle] = None
-            if len(captures) > 0:
-                vehicle_id, trap, mission = rand.choice(captures)
-                vehicle = self._take_existing_vehicle(
-                    sim,
-                    vehicle_id,
-                    agent_id,
-                    mission,
-                    social=agent_id in sim.agent_manager.pending_social_agent_ids,
-                )
-            elif trap.patience_expired(sim.elapsed_sim_time):
+            if ConditionState.EXPIRED in capture.ready_state:
                 # Make sure there is not a vehicle in the same location
```

Comment:
@saulfield The currently failing test has to do with the change in order of the conditions here. I have expired default evaluating first in the exclusive conditions instead of second.

Should I revert the order?

Reply:
I don't like that I changed the order around and am still getting the same result. I can only assume that the generation has changed.

Reply:
Right, it turns out I had changed `zoo_intersection` accidentally.